@xen-orchestra/lite/src/views/pool/PoolNetworkView.vue

```diff
@@ -1,11 +1,200 @@
+                </td>
+                <td>{{ item.name_label }}</td>
+                <td>{{ item.name_description }}</td>
+                <td>{{ item.MTU }}</td>
+                <td>{{ item.default_locking_mode }}</td>
+              </tr>
+            </tbody>
+          </VtsTable>
+          <UiTopBottomTable
+            :selected-items="selectedItemsHostInternalNW"
+            :total-items="reactiveHostPrivateNetworks.length"
+            @toggle-select-all="toggleSelectHostInternalNW"
+          />
+        </div>
+      </div>
+      <UiCardSpinner v-if="!isReady" />
+    </UiCard>
+  </div>
 </template>
 
 <script lang="ts" setup>
-import PageUnderConstruction from '@/components/PageUnderConstruction.vue'
+import UiCard from '@/components/ui/UiCard.vue'
+import UiCardSpinner from '@/components/ui/UiCardSpinner.vue'
 import { usePageTitleStore } from '@/stores/page-title.store'
+import { useNetworkStore } from '@/stores/xen-api/network.store'
+import ColumnTitle from '@core/components/table/ColumnTitle.vue'
+import VtsTable from '@core/components/table/VtsTable.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTableActions from '@core/components/ui/table-actions/UiTableActions.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import {
+  faAlignLeft,
+  faCaretDown,
+  faEdit,
+  faHashtag,
+  faPlus,
+  faPowerOff,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref, watchEffect } from 'vue'
 import { useI18n } from 'vue-i18n'
 
 usePageTitleStore().setTitle(useI18n().t('network'))
+
+const { networksWithVLANs, hostPrivateNetworks, isReady } = useNetworkStore().subscribe()
+
+const reactiveNetworksWithVLANs = ref(networksWithVLANs.value || [])
+const reactiveHostPrivateNetworks = ref(hostPrivateNetworks.value || [])
+
+const selectedItemsNetwork = computed(() => reactiveNetworksWithVLANs.value.filter(item => item.selected).length)
+const selectedItemsHostInternalNW = computed(
+  () => reactiveHostPrivateNetworks.value.filter(item => item.selected).length
+)
+
+const toggleSelectNetwork = (isSelected: any) => {
+  reactiveNetworksWithVLANs.value.forEach(item => {
+    item.selected = isSelected
+  })
+}
+
+const toggleSelectHostInternalNW = (isSelected: any) => {
+  reactiveHostPrivateNetworks.value.forEach(item => {
+    item.selected = isSelected
+  })
+}
+
+watchEffect(() => {
+  if (networksWithVLANs.value) {
+    reactiveNetworksWithVLANs.value = networksWithVLANs.value || []
+  }
+  if (hostPrivateNetworks.value) {
+    reactiveHostPrivateNetworks.value = hostPrivateNetworks.value || []
+  }
+})
```

Comment:
I think this part could be improved by using the _composables_ and utilities for the tables.
See `multi-select.composable.ts` and `table.composable.ts` for example.